superset/views/core.py

```diff
@@ -1785,6 +1786,11 @@ def publish(  # pylint: disable=no-self-use
     @has_access
     @expose("/dashboard/<dashboard_id_or_slug>/")
     @event_logger.log_this_with_extra_payload
+    @check_permissions(
+        on_error=lambda self, ex: Response(
+            utils.error_msg_from_exception(ex), status=403
+        )
+    )
```

Comment:
This is an HTML page. Returning a simple 403 response is probably not enough. We should have a decorator similar to `handle_api_exception` to handle exceptions on HTML pages. 

Ideally all exceptions should be handled in a single final handler for consistent error message rendering.

But maybe the dashboard page is complex enough to warrant its own access denied page, too. 

cc @junlincc

Reply:
I agree
there would be also an effort to move all pages into a single page, maybe it could be part of this effort

Reply:
I understand the need for consistency so feel free to keep it as is if you like but I think at some point it would worth migrating all these to each entity's respective `exceptions.py` and do away from `SupersetSecurityException`.